### PR TITLE
Port is_set_to from TH/THC to ATen.

### PR DESCRIPTION
**Pull request by gchanan**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #28597 Kill _th_zero binding, just use a simple native function instead.
* #28596 Kill THTensor_(match), which isn't used.
* #28595 Write out some set_ overloads instead of relying on code binding generation.
* #28514 Kill zero_dim_tensor_only codegen, it's not used anymore.
* #28513 Remove zero_dim_tensors only from _th_masked_fill_.
* #28512 Remove zero_dim_tensors only from _th_index_fill_.
* #28511 Kill _th_fill binding, which isn't used anymore.
* #28510 unfold: turn off device_guard
* #28475 Port TH/THC implementation of unfold to ATen.
* #28462 Fix scalar handling of unfold.
* #28435 Kill TH(C)Tensor_squeeze which isn't used anymore.
* **#28425 Port is_set_to from TH/THC to ATen.**
* #28422 is_set_to: unify TH/THC implmentation and genericize test_is_set_to.

Differential Revision: [D18063328](https://our.internmc.facebook.com/intern/diff/D18063328)

### CODE REVIEW

File: aten/src/ATen/native/TensorProperties.cpp [hunk 1]
```diff
@@ -81,5 +81,22 @@ Tensor contiguous(const Tensor& self, MemoryFormat memory_format) {
   auto result = at::empty_like(self, self.options(), memory_format);
   return result.copy_(self);
 }
+
+bool is_set_to(const Tensor &self, const Tensor& src) {
```
Comment (mruberry) [inline aten/src/ATen/native/TensorProperties.cpp]: Tensor & vs Tensor&?

File: aten/src/ATen/native/TensorProperties.cpp [hunk 2]
```diff
@@ -81,5 +81,22 @@ Tensor contiguous(const Tensor& self, MemoryFormat memory_format) {
...
+  if (self.storage().unsafeGetStorageImpl() == src.storage().unsafeGetStorageImpl() &&
+      self.storage_offset() == src.storage_offset() &&
+      self.dim() == src.dim())
+  {
+    for (int64_t d = 0; d < self.dim(); ++d)
```
Comment (mruberry) [inline aten/src/ATen/native/TensorProperties.cpp]: for (auto d = decltype(self.dim()){0}; d < ...

File: aten/src/ATen/native/TensorProperties.cpp [hunk 3]
```diff
@@ -81,5 +81,22 @@ Tensor contiguous(const Tensor& self, MemoryFormat memory_format) {
   auto result = at::empty_like(self, self.options(), memory_format);
   return result.copy_(self);
 }
+
```
Comment (mruberry) [inline aten/src/ATen/native/TensorProperties.cpp]: Does this function need a comment? The name isn't super clear.

File: aten/src/ATen/native/TensorProperties.cpp [hunk 4]
```diff
@@ -81,5 +81,22 @@ Tensor contiguous(const Tensor& self, MemoryFormat memory_format) {
...
+bool is_set_to(const Tensor &self, const Tensor& src) {
+  if (self.storage().unsafeGetStorageImpl() == src.storage().unsafeGetStorageImpl() &&
+      self.storage_offset() == src.storage_offset() &&
+      self.dim() == src.dim())
+  {
+    for (int64_t d = 0; d < self.dim(); ++d)
+    {
+      if (self.size(d) != src.size(d) || self.stride(d) != src.stride(d)) {
```
Comment (mruberry) [inline aten/src/ATen/native/TensorProperties.cpp]: Your brace style is alternating between Allman and 1TBS (mandatory braces).

File: aten/src/ATen/native/TensorProperties.cpp [hunk 5]
```diff
@@ -81,5 +81,22 @@ Tensor contiguous(const Tensor& self, MemoryFormat memory_format) {
...
+        return true;
```
Comment (mruberry) [inline aten/src/ATen/native/TensorProperties.cpp]: not return false here?

**Review by mruberry — CHANGES_REQUESTED** [review]
Need to verify return values. Style comments optional (but comments are always nice).

File: aten/src/ATen/native/TensorProperties.cpp (hunk 5 above)
Comment (gchanan) [reply aten/src/ATen/native/TensorProperties.cpp]: right.

**Review by mruberry — APPROVED** [review]
Approving (assuming you'll fix lint).

**Pull request closed without merging**